src/lib.rs

```diff
@@ -154,19 +151,17 @@ pub fn bn128_pairing(input: &[u8], output: &mut BytesRef) -> Result<(), Error> {
 		}
 	};
 
-	let mut buf = [0u8; 32];
-	ret_val.to_big_endian(&mut buf).expect("Cannot fail since 0..32 is 32-byte length");;
-	output.write(0, &buf);
+	ret_val.to_big_endian(&mut *output).expect("Cannot fail since 0..32 is 32-byte length");
```

Comment:
Since `output` is passed as `&mut [u8]`, would `&mut *output` not be equivalent to `output`?

Reply:
oh yeah.. thats true